### PR TITLE
Remove some large copies in ClusterInfo loadPlan via shared_ptr

### PR DESCRIPTION
**Pull request by MBkkt**
### Scope & Purpose

* Remove some large copies in ClusterInfo loadPlan via shared_ptr (I left comments in the PR)
* Replace NodeMap via FlatMap for most cases
* added const to some shared_ptr to avoid occasionally not read-only sharing between threads

- [ ] :hankey: Bugfix
- [ ] :pizza: New feature
- [x] :fire: Performance improvement
- [x] :hammer: Refactoring/simplification

### Checklist

- [ ] Tests
  - [ ] **Regression tests**
  - [ ] C++ **Unit tests**
  - [ ] **integration tests**
  - [ ] **resilience tests**
- [ ] :book: CHANGELOG entry made
- [ ] :books: documentation written (release notes, API changes, ...)
- [ ] Backports
  - [ ] Backport for 3.11: *(Please link PR)*
  - [ ] Backport for 3.10: *(Please link PR)*

#### Related Information

*(Please reference tickets / specification / other PRs etc)*

- [ ] Docs PR: 
- [ ] Enterprise PR: https://github.com/arangodb/enterprise/pull/1371
- [ ] GitHub issue / Jira ticket:
- [ ] Design document:

### CODE REVIEW

File: arangod/Cluster/ClusterInfo.h [hunk 1]
```diff
@@ -214,23 +216,24 @@ class ClusterInfo final {
...
-      AssocUnorderedContainer<pmr::DatabaseID,
-                              std::shared_ptr<DatabaseCollections>>;
+  using DatabaseCollections = FlatMap<pmr::CollectionID, CollectionWithHash>;
+  using AllCollections = FlatMapShared<pmr::DatabaseID, DatabaseCollections>;
+
   using DatabaseCollectionsCurrent =
-      AssocUnorderedContainer<pmr::CollectionID,
-                              std::shared_ptr<CollectionInfoCurrent>>;
-  using AllCollectionsCurrent =
-      AssocUnorderedContainer<pmr::DatabaseID, DatabaseCollectionsCurrent>;
```
Comment (MBkkt) [inline arangod/Cluster/ClusterInfo.h:226]: This was unnecessary copied

File: arangod/Cluster/ClusterInfo.h [hunk 2]
```diff
@@ -214,23 +216,24 @@ class ClusterInfo final {
...
+      FlatMapShared<pmr::CollectionID, CollectionInfoCurrent>;
 
-  using DatabaseViews =
-      AssocUnorderedContainer<pmr::ViewID, std::shared_ptr<LogicalView>>;
-  using AllViews = AssocUnorderedContainer<pmr::DatabaseID, DatabaseViews>;
```
Comment (MBkkt) [inline arangod/Cluster/ClusterInfo.h:230]: This was copied

File: arangod/Cluster/ClusterInfo.h [hunk 3]
```diff
@@ -1162,15 +1159,15 @@ class ClusterInfo final {
...
   AllCollections _plannedCollections;     // from Plan/Collections/
   AllCollections _newPlannedCollections;  // TODO
-  AssocUnorderedContainer<pmr::CollectionID,
-                          std::shared_ptr<std::vector<std::string>>>
+  // TODO is it ok to don't account value for _shards?
+  FlatMapShared<pmr::CollectionID, std::vector<std::string>>
       _shards;  // from Plan/Collections/
                 // (may later come from Current/Collections/ )
   // planned shard => servers map
-  AssocUnorderedContainer<pmr::ShardID, ManagedVector<pmr::ServerID>>
```
Comment (MBkkt) [inline arangod/Cluster/ClusterInfo.h:1170]: This was copied

File: arangod/Cluster/ClusterInfo.h [hunk 4]
```diff
@@ -40,8 +40,6 @@
 #include "Cluster/CallbackGuard.h"
 #include "Cluster/ClusterTypes.h"
 #include "Cluster/RebootTracker.h"
-#include "Containers/NodeHashMap.h"
-#include "Containers/NodeHashSet.h"
```
Comment (jsteemann) [inline arangod/Cluster/ClusterInfo.h:44]: I guess we should include FlatHashSet.h here now.

**Review by jsteemann — APPROVED** [review]
LGTM

**Review by Dronplane — APPROVED** [review]
LGTM

**Pull request merged**